src/WebJobs.Script/Description/ScriptFunctionInvoker.cs

```diff
@@ -70,8 +70,9 @@ public static bool IsSupportedScriptType(ScriptType scriptType)
                     await ExecuteScriptAsync(bashPath, scriptHostArguments, parameters);
                     break;
                 case ScriptType.FSharp:
-                    scriptHostArguments = string.Format("/c fsi.exe \"{0}\"", _scriptFilePath);
-                    await ExecuteScriptAsync("cmd", scriptHostArguments, parameters);
+                    scriptHostArguments = string.Format("\"{0}\"", _scriptFilePath);
+                    string fsiPath = ResolveFSharpPath();
+                    await ExecuteScriptAsync(fsiPath, scriptHostArguments, parameters);
```

Comment:
Downside is that it may now be less flexible on local machines that don't have that exact version? Maybe it should do existence test and fall back to PATH.